### PR TITLE
Make bitcoin-tx replaceable value optional

### LINKED ISSUE #28638: bitcoin-tx replaceable value should be optional, but isn't
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behaviour

The help text says:

```
  replaceable(=N)
       Set RBF opt-in sequence number for input N (if not provided, opt-in all
       available inputs)
```

but when I don't provide N, I get an error:

```
$ bitcoin-tx -create in=0000000000000000000000000000000000000000000000000000000000000000:0 replaceable
error: Invalid TX input index ''
```

### Expected behaviour

I expect bitcoin-tx to act like its help string describes:

```
$ bitcoin-tx -create in=0000000000000000000000000000000000000000000000000000000000000000:0 replaceable
020000000100000000000000000000000000000000000000000000000000000000000000000000000000fdffffff0000000000
```

The fix is simple:

```
diff --git a/src/bitcoin-tx.cpp b/src/bitcoin-tx.cpp
index a05443a2e5..b9348fcad0 100644
--- a/src/bitcoin-tx.cpp
+++ b/src/bitcoin-tx.cpp
@@ -222,7 +222,7 @@ static void MutateTxRBFOptIn(CMutableTransaction& tx, const std::string& strInId
 {
     // parse requested index
     int64_t inIdx;
-    if (!ParseInt64(strInIdx, &inIdx) || inIdx < 0 || inIdx >= static_cast<int64_t>(tx.vin.size())) {
+    if (strInIdx != "" && (!ParseInt64(strInIdx, &inIdx) || inIdx < 0 || inIdx >= static_cast<int64_t>(tx.vin.size()))) {
         throw std::runtime_error("Invalid TX input index '" + strInIdx + "'");
     }
 
```

### Steps to reproduce

Run the command shown above.

### Relevant log output

_No response_

### How did you obtain Bitcoin Core

Compiled from source

### What version of Bitcoin Core are you using?

master@e3eb3aae3

### Operating system and version

Debian 12

### Machine specifications

_No response_

### PR DESCRIPTION
**Pull request by kashifs**
This fixes #28638. The issue was originally raised by dooglus, who also suggested the patch found in this code. Additionally, test coverage has been added and documentation has been updated.

### CODE REVIEW

**Review by pinheadmz — COMMENTED** [review]
concept ACK 605a147b3bf93b7e4325afa4a4df61ffb3523012


I think the test failures are because on CI we copy everything we need for testing into a new directory in the container, and the files you added need to be listed as well in Makefile.md under `EXTRA_DIST`.

For extra context, [this is the command](https://github.com/kashifs/bitcoin/blob/605a147b3bf93b7e4325afa4a4df61ffb3523012/ci/test/03_test_script.sh#L137) that copies the files from `EXTRA_DIST` into the ci scratch/build working directory.

You can run CI locally with docker (see `ci/README.md`). Just for sanity check I entered the test container to prove the files were missing:


```
root@3e18724095c4:/# diff /Users/matthewzipkin/Desktop/work/bitcoin/test/util/data/ /ci_container_base/ci/scratch/build/bitcoin-aarch64-unknown-linux-gnu/test/util/data/
Only in /Users/matthewzipkin/Desktop/work/bitcoin/test/util/data/: txreplace1.hex
Only in /Users/matthewzipkin/Desktop/work/bitcoin/test/util/data/: txreplacenoinputs.hex
Only in /Users/matthewzipkin/Desktop/work/bitcoin/test/util/data/: txreplaceomittedn.hex
Only in /Users/matthewzipkin/Desktop/work/bitcoin/test/util/data/: txreplacesingleinput.hex
```

File: src/bitcoin-tx.cpp [hunk 1]
```diff
@@ -66,7 +66,9 @@ static void SetupBitcoinTxArgs(ArgsManager &argsman)
     argsman.AddArg("outscript=VALUE:SCRIPT[:FLAGS]", "Add raw script output to TX. "
         "Optionally add the \"W\" flag to produce a pay-to-witness-script-hash output. "
         "Optionally add the \"S\" flag to wrap the output in a pay-to-script-hash.", ArgsManager::ALLOW_ANY, OptionsCategory::COMMANDS);
-    argsman.AddArg("replaceable(=N)", "Set RBF opt-in sequence number for input N (if not provided, opt-in all available inputs)", ArgsManager::ALLOW_ANY, OptionsCategory::COMMANDS);
+    argsman.AddArg("replaceable(=N)", "Sets Replace-By-Fee (RBF) opt-in sequence number for input N. "
+        "If N is not provided, the command attempts to opt-in all available inputs for RBF. "
+        "If the transaction has no inputs, no action will be taken by this command. ", ArgsManager::ALLOW_ANY, OptionsCategory::COMMANDS);
```
[the remark this answers is not in the document]
Comment (kashifs) [reply src/bitcoin-tx.cpp:71]: Hi @pinheadmz,
Thanks for taking the time to review this. I've updated the Makefile.

For the documentation, I wanted to make it clear to the end user that if a transaction with no inputs called `tx_noinputs_replaceable` is created by running the following command:`./bitcoin-tx -create replaceable`, it should NOT be assumed that inputs later added to `tx_noinputs_replaceable` will be replaceable.

Is this already obvious? If not, is there other language that I should use to make this clear?

Thanks!

[the remark this answers is not in the document]
Comment (pinheadmz) [reply src/bitcoin-tx.cpp:71]: Ok I see maybe then `"If the transaction has no inputs, this option is ignored"`? is that correct?

**Review by pinheadmz — APPROVED** [review]
ACK 98afe7866185ed4157ffc581763e11dc02fcbae0

Great work Kashif!
Built and ran all tests locally. Confirmed that some new tests fail without the patch, and additional tests were added that expand coverage for the `bitcoin-tx` utility with "replaceable" option. I also played with the updated utility on the command line, and confirmed that this patch closes the issue linked in the PR description. 👍 

<details><summary>Show Signature</summary>

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

ACK 98afe7866185ed4157ffc581763e11dc02fcbae0
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCAAdFiEE5hdzzW4BBA4vG9eM5+KYS2KJyToFAmV5+OwACgkQ5+KYS2KJ
yTqnCBAA4GMxNjFMRiTs1Uw+LX+OXFGBDbnX3Ez2iO/8YRyTGF6kFb5uyRFVXZoC
HByu7oG4ATOND6ge0JENVq7muiJ7odD3qitvUTKU2v6NscOkdDXSyUyxRVj3mnMI
lTrpGCkyS1FbNEvZXr6tbY2wul5HLGzVJ5/B5BA2oNxgYnm2TLB1hu1h4x0h/6Va
xNkpVVVOufRqmxyZtL4wjS8rRQb4zg2vg2Netcfli5mHGEpYYMkv+BDjykz9T0zg
Qs13eJIp78kf2/X0xTMJgwiCdccefx02vFOABBtPT1sHJnh1YrmDRxCOxJFsEDTi
0nJYFZUEz236zKVKzwnQakUFLNHtFDrQiy+VWAcilIfY1gJp534dXzz7XsdKBtN+
ZRqiLsf8aOBHxzetE9j19oV7/J1Nyof0O4dMHZcX7ZkNoH0O7ZpyGg6+9fi46NGP
nDJq+VehUAToJKDTtwGJt5kNumXIlFPdbL2GvGiz/8NI2qUXqBW7RCenWZj5Ht4i
+h847pO4U3A4op6lZRk8dmBdIR7vUeczeMarUeSJDgcoCR2d1oLpySwMiGyEOFLt
vhoXMYo0Ky7958mxYP+ZpsE6M4RuDdvAdP0ofQs0kh+N+Jd7LcuIBkVZsThFhuGm
tMGDyn0jEdC0T8x68B3wZ3h6hXtbpDz4vbsTkYwO04mc53+uA8k=
=vCYW
-----END PGP SIGNATURE-----
```

pinheadmz's public key is [on keybase](https://keybase.io/pinheadmz/pgp_keys.asc?fingerprint=e61773cd6e01040e2f1bd78ce7e2984b6289c93a)
</details>

**Review by hernanmarino — APPROVED** [review]
Tested ACK 98afe7866185ed4157ffc581763e11dc02fcbae0

**Review by instagibbs — APPROVED** [review]
untested ACK https://github.com/bitcoin/bitcoin/pull/29022/commits/98afe7866185ed4157ffc581763e11dc02fcbae0

Test coverage looks good, thanks!

**Comment by achow101** [comment]
ACK 98afe7866185ed4157ffc581763e11dc02fcbae0

**Pull request merged**